Source/JavaScriptCore/runtime/JSONObject.cpp

```diff
@@ -662,15 +662,19 @@ class FastStringifier {
     String firstGetterSetterPropertyName() const;
     void recordFastPropertyEnumerationFailure(JSObject&);
     bool haveFailure() const;
-    unsigned remainingCapacity() const;
+    bool hasRemainingCapacity(unsigned size = 1);
+    bool hasRemainingCapacitySlow(unsigned size);
     bool mayHaveToJSON(JSObject&) const;
 
     static void logOutcome(ASCIILiteral);
     static void logOutcome(String&&);
 
+    static unsigned useableBufferSize(size_t availableBufferSize);
+
     JSGlobalObject& m_globalObject;
     VM& m_vm;
-    unsigned m_length { 0 };
+    unsigned m_length { 0 }; // length of content already filled into m_buffer.
+    unsigned m_useableEnd { 0 };
     bool m_checkedObjectPrototype { false };
     bool m_checkedArrayPrototype { false };
```

Comment:
The comment just below here that says that the `6000` acts a recursion limit is no longer accurate.

Reply:
I'll remove the comment.